channel/src/main/java/com/synopsys/integration/alert/channel/jira/server/util/JiraServerCustomFieldResolver.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * channel
+ *
+ * Copyright (c) 2020 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.channel.jira.server.util;
+
+import com.synopsys.integration.alert.channel.jira.common.JiraCustomFieldResolver;
+import com.synopsys.integration.jira.common.server.service.FieldService;
+
+public class JiraServerCustomFieldResolver extends JiraCustomFieldResolver {
+    public JiraServerCustomFieldResolver(FieldService fieldService) {
+        super(fieldService::getUserVisibleFields);
```

Comment:
Instead of using an entire class for this function, you can open up the method that uses this function to accept the function as a parameter. Then wherever you make the distinction between JiraServerCustomFieldResolver and JiraCloudCustomFieldResolver (Couldn't find it in the PR) you instead set a variable function equal to what is being passed here... does that make sense? I think it would be clearer than having 2 separate basically empty classes.

Reply:
Yeah, I was thinking about this. I didn't know if it would be the most expressive API to ask for a `Supplier` in the constructor, but I think you're right that it's better than the duplication.